src/shogun/distributions/KernelDensity.cpp

```diff
@@ -91,12 +91,14 @@ bool KernelDensity::train(std::shared_ptr<Features> data)
 	return true;
 }
 
-SGVector<float64_t> KernelDensity::get_log_density(const std::shared_ptr<DenseFeatures<float64_t>>& test, int32_t leaf_size)
+SGVector<float64_t> KernelDensity::get_log_density(const std::shared_ptr<Features>& test, int32_t leaf_size)
 {
 	require(test,"data not supplied");
-
+	std::shared_ptr<DenseFeatures<float64_t>> dense_feat =std::dynamic_pointer_cast<DenseFeatures<float64_t>>(test);
```

Comment:
`auto dense_feat = ...` you already show what the type is in the cast

Reply:
True

Reply:
Done